core/trino-main/src/main/java/io/trino/server/DynamicFilterService.java

```diff
@@ -719,8 +721,10 @@ private void collectReplicated(Domain domain)
 
         private void collectPartitioned(TaskId taskId, Domain domain)
         {
-            if (!collectedTasks.add(taskId.getPartitionId())) {
-                return;
+            synchronized (collectedTasks) {
+                if (!collectedTasks.checkedAdd(taskId.getPartitionId())) {
```

Comment:
Since we know the expected tasks count before collection begins, could we just create a dense bitmap ourselves ? Roaring could still be more compact due to using RLE, but I'm not sure if we really need to squeeze space that much here.

Reply:
The expected number of tasks is supplied later on asynchronously. Also while implementing an efficient bitmap is not extremely difficult it is also non trivial, thus I would probably stick with the library.